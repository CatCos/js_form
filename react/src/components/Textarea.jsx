import React, { PropTypes } from 'react';

function handleChange(e, inputName, props) {
  if (props.onChange) {
    return props.onChange({
      e: e,
      inputName: inputName
    });
  }
}

/**
 * Render function
 *
 * @params props Component properties
 */
function Textarea(props) {
  return (
    <div className="form-group">
      <label htmlFor={props.name} className="label">{props.text}</label>
      <textarea
        className="textarea"
        id={props.name}
        type={props.type}
        value={props.value}
        name={props.name}
        onChange={(e) => handleChange(e, props.field, props)}/>
    </div>
  );
}

export default Textarea;
