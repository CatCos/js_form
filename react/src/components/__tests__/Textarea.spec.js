import React from 'react';
import Textarea from '../Textarea';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Textarea
      value="text"
      name='text'
      type='text'
      field='text'
      onChange={jest.fn()}
      text='Submit comment'/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
