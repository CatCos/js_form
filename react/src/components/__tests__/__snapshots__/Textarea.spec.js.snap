// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="form-group"
>
  <label
    className="label"
    htmlFor="text"
  >
    Submit comment
  </label>
  <textarea
    className="textarea"
    id="text"
    name="text"
    onChange={[Function]}
    type="text"
    value="text"
  />
</div>
`;
